### ISSUE TITLE
After upgrade to nginx 1.22.0 no stream is delivered anymore

### ISSUE THREAD

**Issue by Kulturserver** [arut/nginx-rtmp-module#1703]
With the nginx-version 1.17.9 the module is running just fine. When updating to the nginx version 1.22.0, it compiles and can be started with the module, but no stream is delivered anymore, it is only received. Any suggestions what might be the issue or is this a known problem? Any help would be appreciated, thanks!

**Comment by valerio-bozzolan** [comment]
Can you share something from your log files?

Can you share your minimal config to reproduce the problem?

**Comment by Kulturserver** [comment]
Sure - here is the logfile, one irritating thing is, that we would expect to see the stream being initiated or something similar

::ffff:162.55.76.22 [24/Oct/2022:09:05:34 +0200] PUBLISH "live" "test-wkn" "" - 30126944 759 "" "FMLE/3.0 (compatible; FMSc/1.0)" (44s)
::ffff:89.245.220.150 [24/Oct/2022:09:05:59 +0200] PLAY "live" "test-wkn" "" - 390 436 "" "LNX 9,0,124,2" (52s)
::ffff:89.245.220.150 [24/Oct/2022:09:06:29 +0200] PLAY "live" "test-wkn" "" - 390 436 "" "LNX 9,0,124,2" (22s)
::ffff:162.55.76.22 [24/Oct/2022:09:06:41 +0200] PUBLISH "live" "test-wkn" "" - 38226467 759 "" "FMLE/3.0 (compatible; FMSc/1.0)" (56s)

Our config: 

server        {
    listen      [::]:1935;
    chunk_size  4096;
    application live    {
      live      on;
      access_log        /var/log/nginx/rtmp-access.log;
      hls       on;
      hls_path  /var/www-data/live_hls;
      hls_nested        on;
      hls_fragment_naming       system;
      hls_fragment      10s;
      hls_cleanup       off;
      hls_continuous    on;
      allow     publish all;
      allow     play    all;
      exec_publish_done /var/www-data/bin/nginx_exec_publish_done.sh "live_hls" "$name";
    }
}

**Comment by marianocarpentier** [comment]
Did you find the issue? I'm having the same problem.

**Comment by Kulturserver** [comment]
No, unfortunately not - was thinking of updating this thread today as well...

**Comment by marianocarpentier** [comment]
My problem turned out to be different after all. It wasn't the upgrade but the fact I was running the ffmpeg command without sudo. Although initially that worked, it didn't anymore after restarting that server.

**Comment by Docteh** [comment]
Since you've got HLS turned on, you can look in `hls_path /var/www-data/live_hls;` for HLS segments when the stream is being delivered. 

Just as a curiousity, what are you using as an RTMP client?

Personally I compiled this today, and since I use OBS RTMP -> nginx -> HLS player for clients, I didn't see a bug.

**Comment by Kulturserver** [comment]
Thanks for replying!

> My problem turned out to be different after all. It wasn't the upgrade but the fact I was running the ffmpeg command without sudo. Although initially that worked, it didn't anymore after restarting that server.

Our problem actually seems to depend on ffmpeg, respectively its execution - if we disable ffmpeg via exec_push the stream works with the newest version, but as soon as we enter a parameter to exec_push the stream isn't delivered to the newer versions anymore.

**Comment by Kulturserver** [comment]
it seems that the config above wasn't the actual one, we've got HLS turned off - sorry for that! Here's the current settings:

rtmp	{
  server	{
    listen	[::]:1935;
    chunk_size	4096;

    # no hls, just further processing
    application feed    {
      live	on;
      allow	publish	all;
      allow	play	all;
      access_log	/var/log/nginx/rtmp.log;
      exec_kill_signal  15;
#      exec_push		sudo /usr/bin/ffmpeg -nostdin -re -i rtmp://localhost/feed/$name
#			-f flv -c:v copy -c:a copy rtmp://server.domain.org/record/${name}_feed;
    }

**Comment by tw0517tw** [comment]
Hi @Kulturserver , I also faced this problem but I found in #1719 that adding `multi_accept on` can fix the problem if you are using `worker_processes auto;` or anything not `1`

```
events {
    multi_accept on;
}
```
